api/credentials_controller.go

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api
+
+import (
+	"context"
+	"fmt"
+	"github.com/Peripli/service-manager/api/osb"
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/gofrs/uuid"
+	"github.com/tidwall/gjson"
+	"net/http"
+	"time"
+
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/web"
+)
+
+// CredentialsController implements api.Controller by providing logic for broker platform credential storage/update
+type CredentialsController struct {
+	*BaseController
+}
+
+func NewCredentialsController(ctx context.Context, options *Options) *CredentialsController {
+	return &CredentialsController{
+		BaseController: NewController(ctx, options, web.BrokerPlatformCredentialsURL, types.BrokerPlatformCredentialType, func() types.Object {
+			return &types.BrokerPlatformCredential{}
+		}),
+	}
+}
+
+func (c *CredentialsController) Routes() []web.Route {
```

Comment:
what if multiple calls to this api are done and applying the notification result by the agent in CC fails ?
From a high level i think that the last known credentials in CC should be still valid in SM

Reply:
We've discussed ways in which to keep the last valid credentials in SM, like for example - not allowing rotation of credentials, if the current ones have not been used yet (old_username & old_password are not ""), but there are numerous bad consequences of this like for example we'll never be able to rotate credentials ever again.